gunicorn/sock.py

```diff
@@ -20,6 +20,7 @@
 class BaseSocket(object):
 
     def __init__(self, address, conf, log, fd=None):
+        self.parent = os.getpid()
```

Comment:
`self.parent` should just be removed. It's not doing anything anymore. It is always set to `os.getpid()` so it will always be equal to `os.getpid()` at close. This is because after USR2 exec these objects are still recreated in the new process.

If the intention is to prevent the destruction of systemd activated sockets then we need a separate flag here that we pass to the constructor only when called in the `LISTEN_FDS` loop, or we need to check for the environment variable here (a parameter is easier to unit test).

The arbiter is now tracking whether it's the "master" arbiter and intelligently calls `close` (or not), so the only information we need here is whether Gunicorn (any process) created the socket or whether the init system did.

It's not actually important which PID created the socket (the arbiter tracks whether to call `close()`)